File: assesment_nazmul/assesment_nazmul/report/procurement_report/procurement_report.py
# Copyright (c) 2026, Nazmul Hossain and contributors
# For license information, please see license.txt

import frappe
from frappe import _


def execute(filters: dict | None = None):
	if not filters:
		filters = {}

	columns = get_columns()
	data = get_data(filters)

	return columns, data


def get_columns() -> list[dict]:
	return [
		{
			"label": _("Requisition Number"),
			"fieldname": "requisition_no",
			"fieldtype": "Link",
			"options": "Procurement Requisition",
			"width": 180,
		},
		{
			"label": _("Requested By"),
			"fieldname": "requested_by",
			"fieldtype": "Link",
			"options": "Employee",
			"width": 160,
		},
		{
			"label": _("Department"),
			"fieldname": "department",
			"fieldtype": "Data",
			"width": 150,
		},
		{
			"label": _("Estimated Budget"),
			"fieldname": "estimated_total_amount",
			"fieldtype": "Currency",
			"width": 140,
		},
		{
			"label": _("Current Status"),
			"fieldname": "status",
			"fieldtype": "Data",
			"width": 130,
		},
		{
			"label": _("Workflow Status"),
			"fieldname": "workflow_state",
			"fieldtype": "Link",
			"options": "Workflow State",
			"width": 140,
		},
	]


def get_data(filters: dict) -> list[dict]:
	conditions = {}
	if filters.get("department"):
		conditions["department"] = filters.get("department")

	if filters.get("status"):
		conditions["status"] = filters.get("status")

	return frappe.db.get_all(
		"Procurement Requisition",
		fields=["name as requisition_no", "requested_by", "department", "estimated_total_amount", "status", "workflow_state"],
		filters=conditions,
		order_by="creation desc",
	)

File: assesment_nazmul/assesment_nazmul/report/procurement_report/procurement_report.js
// Copyright (c) 2026, Nazmul Hossain and contributors
// For license information, please see license.txt

frappe.query_reports["Procurement Report"] = {
	filters: [
		{
			fieldname: "department",
			label: __("Department"),
			fieldtype: "Link",
			options: "Department",
		},
		{
			fieldname: "status",
			label: __("Status"),
			fieldtype: "Select",
			options: "\nDraft\nPending Approval\nApproved\nRejected\nCancelled\nOrdered",
		},
	],
	formatter: function(value, row, column, data, default_formatter) {
		value = default_formatter(value, row, column, data);
		if (data && (column.fieldname === "status" || column.fieldname === "workflow_state")) {
			let status_val = data[column.fieldname];
			if (!status_val) return value;

			let color = "gray";
			if (["Approved", "Ordered"].includes(status_val)) {
				color = "green";
			} else if (["Pending Approval", "Waiting For RFQ"].includes(status_val)) {
				color = "orange";
			} else if (["Rejected", "Cancelled"].includes(status_val)) {
				color = "red";
			} else if (status_val === "Draft") {
				color = "blue";
			}
			return `<span class="indicator-pill ${color}">${status_val}</span>`;
		}
		return value;
	},
};
